api/Api/Aether/Lobby/LobbyClient.cs

```diff
@@ -143,6 +144,38 @@ private class Inner : IDisposable {
                 }
             }
 
+            public Task<Result<LobbyMessage?, Exception>> UploadRobotData(DataRobot robot) {
+                if (!_isAlive.Value)
+                    return Task.FromResult(new Result<LobbyMessage?, Exception>(new Exception("Client no longer alive")));
+
+                var request = new LobbyMessage.Types.ToDataRobot {
+                    Guid = _handler.Guid,
+                    DataRobot = robot
+                };
+
+                var task = new Task<Result<LobbyMessage?, Exception>>(() => {
+                    var response = HandleResponseBoilerplate(new LobbyMessage { ToDataRobot = request });
+                    if (response.isError) {
+                        return response;
+                    }
+
+                    var msg = response.GetResult()!;
+                    switch (msg.MessageTypeCase) {
+                        case LobbyMessage.MessageTypeOneofCase.FromDataRobot:
+                            // TODO: Accept needed robot data from server.
```

Comment:
This will be done as part of [JIRA Issue - 1516](https://jira.autodesk.com/browse/AARD-1516).